array/builder.gen.go

```diff
@@ -71,10 +71,16 @@ func (b *IntBuilder) NewArray() Array {
 func (b *IntBuilder) NewIntArray() *Int {
 	return b.b.NewInt64Array()
 }
-func (b *IntBuilder) CopyValidValues(values *Int) {
-	b.Reserve(values.Len() - values.NullN())
+func (b *IntBuilder) CopyValidValues(values *Int, nullCheckArray Array) {
+	if values.Len() != nullCheckArray.Len() {
+		panic("Length mismatch between the value array and the null check array")
+	}
+	b.Reserve(values.Len() - nullCheckArray.NullN())
+
+	nullBitMapBytes := nullCheckArray.NullBitmapBytes()
+	nullOffset := nullCheckArray.Data().Offset()
 	for i := 0; i < values.Len(); i++ {
-		if values.IsValid(i) {
+		if isValid(nullBitMapBytes, nullOffset, i) {
 			b.Append(values.Value(i))
```

Comment:
I don't think this is quite right, since you want to call `appendNull` here if the values array contains a `null`. What you have here will append a (valid) arbitrary value (probably the zero value).
```suggestion
			if values.IsValid(i) {
			    b.Append(values.Value(i))
			} else {
			    b.AppendNull()
			}
```

Reply:
Yeah, I were just thinking of the time column which never has any nulls so I didn't handle that case.